File: javascript/exercise45/word-finder.js
import fs from "node:fs";
import { input } from "pynput";

const input_file = "./input.txt";
const contents = fs.readFileSync(input_file, "utf8");

let output_file = "./output.txt";

const output = await input("Enter output file: ");
if (output.trim() !== "") {
	output_file = output;
}

const replaced = contents.replace(/utilize/g, "use");

fs.writeFileSync(output_file, replaced);

File: go/word-finder/word_finder.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

const inputFile = "./input.txt"

var outputFile = "./output.txt"

func main() {
	output := scanInput("Enter output file: ")
	if output != "" {
		outputFile = output
	}

	file, err := os.ReadFile(inputFile)
	if err != nil {
		fmt.Println(err)
		return
	}

	strings.Count("s string", "")
	replaced := strings.ReplaceAll(string(file), "utilize", "use")

	os.WriteFile(outputFile, []byte(replaced), os.ModePerm)
}

func scanInput(prompt string) string {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print(prompt)
	scanner.Scan()

	return strings.TrimSpace(scanner.Text())
}
